bolt-sidecar/src/api/commitments/firewall/receiver.rs

```diff
@@ -0,0 +1,616 @@
+use alloy::signers::local::PrivateKeySigner;
+use futures::StreamExt;
+use std::{
+    fmt::{self, Debug, Formatter},
+    time::Duration,
+};
+use thiserror::Error;
+use tokio::sync::{broadcast, mpsc};
+use tokio_tungstenite::{
+    connect_async_with_config,
+    tungstenite::{self, client::IntoClientRequest, protocol::WebSocketConfig},
+};
+use tracing::{error, info};
+
+use reqwest::Url;
+
+use crate::{
+    api::commitments::server::CommitmentEvent,
+    common::{
+        backoff::{retry_with_backoff_if, RetryConfig},
+        secrets::EcdsaSecretKeyWrapper,
+    },
+    config::{chain::Chain, limits::LimitsOpts},
+    primitives::misc::ShutdownSignal,
+};
+
+use super::{
+    jwt::ProposerAuthClaims,
+    processor::{CommitmentRequestProcessor, InterruptReason, ProcessorState},
+};
+
+/// The interval at which to send ping messages from connected clients.
+#[cfg(test)]
+const PING_INTERVAL: Duration = Duration::from_secs(4);
+#[cfg(not(test))]
+const PING_INTERVAL: Duration = Duration::from_secs(30);
+
+/// The maximum number of retries to attempt when reconnecting to a websocket server.
+const MAX_RETRIES: usize = 1000;
+
+/// The maximum messages size to receive via websocket connection, in bits, set to 32MiB.
+///
+/// It is enough to account for a commitment request with 6 blobs and the largest
+/// memory-consuming transactions you can create. Reference: https://xn--2-umb.com/22/eth-max-mem/
+const MAX_MESSAGE_SIZE: usize = 16 << 23;
```

Comment:
isn't 16 x 2^23 equal to 128MiB?

Reply:
You're right, `16 << 23` is 128 MiB. I don't remember why I shifted it by 23, writing `32 << 20` is much simpler. Going to change it